x/gamm/twap/store_test.go

```diff
@@ -0,0 +1,181 @@
+package twap_test
+
+import (
+	"time"
+
+	"github.com/osmosis-labs/osmosis/v10/x/gamm/twap/types"
+)
+
+// TestTrackChangedPool takes a list of poolIds as test cases, and runs one list per block.
+// Every simulated block, checks that there no changed pools.
+// Then runs k.trackChangedPool on every item in the test case list.
+// Then checks that changed pools returns the list, deduplicated.
+//
+// This achieves testing the functionality that we depend on, that this clears every end block.
```

Comment:
Nor sure if this is already documented elsewhere but I think we should highlight in README that transient store is what enables these to be cleared every block. Might not be an obvious thing to inexperienced sdk devs

Reply:
Adding to README in the main PR

Reply:
Added!